src/view/ui.py

```diff
@@ -21,19 +21,16 @@ def menuFromDict(acts, activation_function=None, menu=None):
     """
     if menu is None:
         menu = QtWidgets.QMenu()
-    for a in acts:
-        if a is None:
-            menu.addSeparator()
-        elif isinstance(a, dict):
-            for suba, subacts in a.items():
-                submenu = menu.addMenu(suba)
-                menuFromDict(subacts, activation_function, submenu)
-        else:
+    for a, subacts in acts.items():
+        if len(subacts) == 0:
             act = menu.addAction(a)
             if activation_function is not None:
                 def connect(action, s):
                     action.triggered.connect(lambda: activation_function(action))
                 connect(act, a)
```

Comment:
You did not modified this code but I do not understand why you defined the function that you use just after defining, is it not possible to just use the line in the function directly?

Reply:
If I do not declare the connection inside a function with its own argument, the argument 'action' will be updated with the next iteration of the for-loop. And I cannot create a simple copy of a QAction object. So it is the only way I found to work around the problem..

For example: replace this part of code with 
`act.triggered.connect(lambda: activation_function(act))`
then create a load node
and create a child 'add' node on it, it will create a 'divide' node instead because the argument 'act' were updated in the for loop!

Reply:
Oh ok, I see. The ultimate question that I have is, what is the purpose of the `s` parameter of your function `connect()`? You do not even use it.

Reply:
Yep I forgot to remove the argument when I updated this part of code. I removed it in the last commit "update check empty list"!